tests/scenario/test_machine_charm/test_relation_priority.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import json
+from unittest.mock import patch
+
+import pytest
+from charms.grafana_agent.v0.cos_machine import COSMachineProvider
+from scenario import Relation, State
+from scenario import trigger as _trigger
+
+import grafana_agent
+import machine_charm
+from tests.scenario.helpers import CHARM_ROOT, get_charm_meta
+from tests.scenario.test_machine_charm.helpers import set_run_out
+
+
+def trigger(evt: str, state: State, **kwargs):
+    return _trigger(
+        event=evt,
+        state=state,
+        charm_type=machine_charm.GrafanaAgentMachineCharm,
+        meta=get_charm_meta(machine_charm.GrafanaAgentMachineCharm),
+        copy_to_charm_root={"/src/": CHARM_ROOT / "src"},
+        **kwargs,
+    )
+
+
+@pytest.fixture
+def mock_cfg_path(tmp_path):
+    return tmp_path / "foo.yaml"
+
+
+@pytest.fixture(autouse=True)
+def patch_all(mock_cfg_path):
+    grafana_agent.CONFIG_PATH = mock_cfg_path
```

Comment:
This is a monkeypatch.
Did you mean: `patcher = patch('grafana_agent.CONFIG_PATH', mock_cfg_path)`?

Reply:
is that better?

Reply:
Less unexpected side-effects, although here in tests this is unlikely to matter.